### PR TITLE
Issue13-Unit testing

### PR DESCRIPTION
**Pull request by ArCiGoUnoSquare**
Unit test cases for `<Grid />` component

### CODE REVIEW

**Comment by geoperez** [comment]
You got conflicts 😸

**Comment by ArCiGoUnoSquare** [comment]
Disadvantages of turning off the notifications

**Comment by kadosh** [comment]
Can we get this done today?

File: test/Grid.spec.js [hunk 1]
```diff
@@ -247,6 +247,39 @@ describe('<Grid />', () => {
...
+    
+      const grid = <Grid dataSource = { new RemoteDataSource('http://tubular.azurewebsites.net/api/orders/paged') }
+        columns = { columns }
+        showBottomPager = { true }
+      />;
+
+      const wrapper = shallow(grid);
+      const rowFooter = wrapper.find(Table).find(TableFooter).find(TableRow);
+
+      expect(rowFooter).to.have.lengthOf(1);
```
Comment (kadosh) [inline test/Grid.spec.js]: This is not checking if there's a paginator inside, it is just checking if there's a row.

File: test/Grid.spec.js [hunk 2]
```diff
@@ -247,6 +247,39 @@ describe('<Grid />', () => {
...
+      const grid = <Grid dataSource = { new RemoteDataSource('http://tubular.azurewebsites.net/api/orders/paged') }
+        columns = { columns }
+        showBottomPager = { true }
+        footerRenderer = { footerRenderer }
+      />;
+
+      const wrapper = shallow(grid);
+      const rowFooter = wrapper.find(Table).find(TableFooter).find(TableRow);
+
+      expect(rowFooter).to.have.lengthOf(2);
```
Comment (kadosh) [inline test/Grid.spec.js]: Same here, if the test is making sure a footer and a paginator are there, then do the check

File: test/Grid.spec.js [hunk 3]
```diff
@@ -298,6 +299,26 @@ describe('<Grid />', () => {
       expect(rowFooter).to.have.lengthOf(1);
     });
   });
+  
+  describe('RemoteDataSource', () => {
+    it('should response with status 200 when column structure is OK', () => {
+      mock.onPost('http://tubular.azurewebsites.net/api/orders/paged', { columns }).reply(200);
+
+      axiosInstance.post('http://tubular.azurewebsites.net/api/orders/paged', { columns })
```
Comment (kadosh) [inline test/Grid.spec.js]: axiosInstance again??? NOOOOOOOOOOOO

File: test/Grid.spec.js [hunk 4]
```diff
@@ -298,6 +299,26 @@ describe('<Grid />', () => {
...
+        .then(response => {
+          expect(response.status).to.equal(200);
+        });
+    });
+
+    it('should response with status 500 when column structure is not OK', () => {
+      mock.onPost('http://tubular.azurewebsites.net/api/orders/paged', { fakeColumnStructure }).reply(500);
+
+      axiosInstance.post('http://tubular.azurewebsites.net/api/orders/paged', { fakeColumnStructure })
```
Comment (kadosh) [inline test/Grid.spec.js]: The tests shouldn't be calling axiosInstance methods, please read carefully the docs.

File: test/Grid.spec.js (hunk 3 above)
Comment (ArCiGoUnoSquare) [reply test/Grid.spec.js]: https://media.giphy.com/media/RFDXes97gboYg/giphy.gif

**Comment by kadosh** [comment]
Failing....

**Pull request merged**